src/gnn_tracking/graph_construction/graph_builder.py

```diff
@@ -68,7 +68,7 @@ def __init__(
             [1000.0, np.pi, 1000.0, 1, 1 / 1000.0, 1 / 1000.0]
         )
         self._data_list = []
-        self.outfiles = os.listdir(outdir)
```

Comment:
Even better would be to completely remove this attribute and go with `self.outdir.iterdir()` where necessary  (which would probably remove some of the `/` joins below). But that's a little bit more refactoring

Reply:
Let me know if you want to do this

Reply:
Maybe this would not reduce the number of joins since they are for two different directories, but I guess it would reduce the number of string splits.

Can you check if you meant something like [c2eedc1](https://github.com/gnn-tracking/gnn_tracking/pull/268/commits/c2eedc1df22ede61f1c3c95e2c3e5bd302a21025)?